### PR TITLE
feat(inputs.chrony): Remove chronyc dependency

### PR DESCRIPTION
**Pull request by srebhan**
## Summary

This PR removes the dependency for `chronyc` to be installed in preparation of tackling #2798. That change now also allows to check remote chrony daemons if they are configured to allow this (see `bindcmdaddress` and `cmdallow` of [chronyd documentation](https://chrony-project.org/doc/4.5/chrony.conf.html)).

Furthermore, we do add an integration test against a chrony docker container and improve the non-integration test.

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

related to #2798

### CODE REVIEW

File: plugins/inputs/chrony/chrony.go [hunk 1]
```diff
@@ -5,125 +5,179 @@ import (
...
+			// Check if we do have a port and add the default port if we don't
+			if u.Port() == "" {
+				u.Host += ":323"
+			}
+			// We cannot have path elements in an UDP address
+			if u.Path != "" {
+				return fmt.Errorf("path detected in UDP address %q", c.Server)
+			}
+			u = &url.URL{Scheme: "udp", Host: u.Host}
+		case "":
```
Comment (powersj) [inline plugins/inputs/chrony/chrony.go:59]: I'd rather you fall the default case here and require the scheme for the user to be explicit about a scheme. 

Are you trying to parse values like `server:323`? If so, url.Parse, will report the scheme as 'server'. The only way to have an empty string for the scheme is if the user uses `//server:385`.

**Review by powersj — COMMENTED** [review]
This is pretty cool, thanks for the integration test too!

File: plugins/inputs/chrony/chrony.go (hunk 1 above)
Comment (srebhan) [reply plugins/inputs/chrony/chrony.go:59]: Nice catch. Will change to require an explicit scheme.

**Review by powersj — APPROVED** [review]
Landing after playing with this locally.

**Pull request merged**